sdk/spring/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/convert/MappingCosmosConverter.java

```diff
@@ -142,12 +165,83 @@ public JsonNode writeJsonNode(Object sourceEntity) {
             final String id = value == null ? null : value.toString();
             cosmosObjectNode.put("id", id);
         }
+        if (transientFields != null && !transientFields.isEmpty()) {
+            // Strip fields with @Transient annotation from the JsonNode so they are not persisted.
+            cosmosObjectNode = stripTransientFields(cosmosObjectNode, transientFields);
+        }
 
         mapVersionFieldToEtag(sourceEntity, cosmosObjectNode);
 
         return cosmosObjectNode;
     }
 
+    /**
+     * To repopulate any transient fields that were stripped from the original object
+     * @param <T> type of source entity
+     * @param transientValues transient fields and their values
+     * @param responseItem must not be {@literal null}
+     * @return CosmosItemProperties
+     * @throws MappingException no mapping metadata for entity type
+     * @throws CosmosAccessException fail to map document value
+     */
+
+    public <T> JsonNode repopulateAnyTransientFieldsFromMap(JsonNode responseItem, HashMap<Field, Object> transientValues) {
+        if (transientValues.isEmpty()) {
+            return responseItem;
+        }
+        ObjectNode updatedItem = (ObjectNode) responseItem;
+        for (Field field : transientValues.keySet()) {
+            Object value = transientValues.get(field);
+            if (value != null) {
+                updatedItem.set(field.getName(), objectMapper.valueToTree(value));
+            }
+        }
+        return updatedItem;
+
+    }
+
+    /**
+     * To get transient fields and their values
+     * @param <T> type of source entity
+     * @param object must not be {@literal null}
+     * @param transientFields transient fields
+     * @return HashMap
+     */
+    public <T> HashMap<Field, Object> getTransientFieldsAndValuesMap(T object, List<String> transientFields) {
+        HashMap<Field, Object> transientValuesMap = new HashMap<>();
+        for (String fieldName : transientFields) {
+            Field field = null;
+            try {
+                field = object.getClass().getDeclaredField(fieldName);
+                field.setAccessible(true);
+                transientValuesMap.put(field, field.get(object));
+            } catch (NoSuchFieldException | IllegalAccessException e) {
+                LOGGER.info("No value found for transient field: " + fieldName + " with error: " + e.getMessage());
```

Comment:
why log as info if this is an error?
Also, in production scenarios, most of the times customers will not log info. So in this case, we will silently ignore the error.
It is worth brainstorming whether to swallow this error or to break the code flow and throw the error.

Reply:
Yes, the goal was to swallow the exception. We could throw it, but that would break other tests because they are not setting transient field values. Maybe worth discussing what is best to do.

Reply:
Changed to throw exception.